prelude/src/core/language/Iterator.sk

```diff
@@ -69,6 +69,17 @@ mutable base class .Iterator<+T> {
     mutable MapIterator(this, f)
   }
 
+  // i.mapWhile(f) is similar to i.map(f).takeWhile(isSome).map(Option.get)
+  // but terminates as soon as f returns None()
```

Comment:
wouldn't the above also stop mapping as soon as f returns none? Is map not lazy on an iterator?

Reply:
Indeed. I simplified down to this and didn't pay attention that in the end it wasn't needed at all, thanks!

Reply:
It is lazy but if you `.collect(Array)` then `i.map(f).takeWhile(isSome).map(Option.get)` will still go through till the end of `i`

Reply:
I don't think I understand, can you expand on this? Do you mean that `i.map(f).takeWhile((x) -> x.isSome()).collect(Array)` will call `f` on elements of `i` after a `None()` is produced? The implementation of `takeWhile` is:
```
  overridable mutable fun takeWhile(p: T -> Bool): mutable Iterator<T> {
    for (v in this) {
      if (!p(v)) {
        break void
      } else {
        yield v
      }
    }
  }
```
While I haven't found docs on specifically what `break` and `yield` mean, my understanding of this is that the iterator produced by `takeWhile` will be shorter than its argument if an element that does not satisfy `p` is found.

Reply:
From time to time it happens.

Reply:
I liked `mapWhile` because it doesn't use `Object.get` which may throw (though won't, but could, but won't, but could).

And as we found out, laziness is quite nasty: https://github.com/SkipLabs/skjs/issues/88